pkg/dockerregistry/client.go

```diff
@@ -250,7 +254,30 @@ func (c *connection) ImageByTag(namespace, name, tag string) (*Image, error) {
 		return nil, err
 	}
 
-	return repo.getTaggedImage(c, searchTag, tag)
+	image, _, err := repo.getTaggedImage(c, searchTag, tag)
+	return image, err
+}
+
+// ImageManifest returns raw manifest of the specified image within the named Docker image repository
+func (c *connection) ImageManifest(namespace, name, tag string) (string, []byte, error) {
+	if len(namespace) == 0 && imageapi.IsRegistryDockerHub(c.url.Host) {
+		namespace = imageapi.DockerDefaultNamespace
+	}
+	if len(name) == 0 {
+		return "", nil, fmt.Errorf("image name must be specified")
+	}
+	searchTag := tag
+	if len(searchTag) == 0 {
+		searchTag = imageapi.DefaultImageTag
+	}
+
+	repo, err := c.getCachedRepository(fmt.Sprintf("%s/%s", namespace, name))
+	if err != nil {
+		return "", nil, err
+	}
+
+	image, manifest, err := repo.getTaggedImage(c, searchTag, tag)
```

Comment:
in case the error is returned, I guess the image will be nil so the code below will panic?